pkg/provisioner/ironic/ironic.go

```diff
@@ -139,14 +139,15 @@ func (p *ironicProvisioner) validateNode(ironicNode *nodes.Node) (errorMessage s
 func (p *ironicProvisioner) listAllPorts(address string) ([]ports.Port, error) {
 	var allPorts []ports.Port
 
-	opts := ports.ListOpts{
-		Fields: []string{"node_uuid"},
+	// Don't call the ironic api when the port mac address is empty.
```

Comment:
Under what conditions do we provide an empty MAC address here? If it's a user error, shouldn't this call return an error?

Reply:
When we can't find a node by name, we look for it by port. The address is the bootMACAddress, which is optional. So I think returning an empty list is indeed the right thing here.

Reply:
Honestly, I'd rather not call this function with empty address.

Reply:
I will update this commit.